include/picongpu/particles/ParticlesInit.kernel

```diff
@@ -152,7 +152,7 @@ namespace picongpu
 
                     if(realParticlesPerCell > 0.0_X)
                         numParsPerCell
-                            = posFunctor.template numberOfMacroParticles<ParticleType>(realParticlesPerCell);
+                            = posFunctor.template numberOfMacroParticles<ParticleType>(realParticlesPerCell, cellIdx);
```

Comment:
line 255

```
particle[localCellIdx_] = idx;
```
is avoiding that you can spawn particles outside of the cell the algorithm is selecting.